### PR TITLE
added remove and context save

### PR DESCRIPTION
**Pull request by nirvik**

### CODE REVIEW

File: registry/cloudflare-workers/index.js [hunk 1]
```diff
@@ -26,44 +37,34 @@ const doWorkerDeploy = async ({ zone_id, script_path }) => {
...
+  return response
 }
-/*
-const remove = (input, context) => {
-	context.log('Removing worker script ')
-	await removeWorker(input)
-	context.log('awesome ')
+
+const remove = async (input, context) => {
+  context.log('removing!')
```
Comment (pinkerton) [inline registry/cloudflare-workers/index.js]: Do other components leave in these logging statements?

File: registry/cloudflare-workers/serverless.yml [hunk 2]
```diff
@@ -9,3 +9,20 @@ inputTypes:
         script_path:
                 type: string
                 required: true
+
+
+outputTypes:
```
Comment (pinkerton) [inline registry/cloudflare-workers/serverless.yml]: Is this using the same tab style as the rest of the components repo?

**Review by pinkerton — APPROVED** [review]
Looks great! 

A couple small nits, but I defer to you, the expert 😄

File: registry/cloudflare-workers/index.js (hunk 1 above)
Comment (nirvik) [reply registry/cloudflare-workers/index.js]: I think they do.. Check this out: https://github.com/serverless/components/blob/master/registry/netlify-site/src/index.js#L159

File: registry/cloudflare-workers/index.js [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
+}
+
+const deploy = async (input, context) => {
+  context.log('Deploying worker script')
+  const { success, errors, messages, result } = await doWorkerDeploy(input)
+
+  if (!success) {
+    errors.forEach((err) => {
+      let { message } = err
+      context.log(message)
```
Comment (EverlastingBugstopper) [inline registry/cloudflare-workers/index.js]: nit: why don't you just `context.log(err)` here?

File: registry/cloudflare-workers/index.js [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
+    })
+  } else {
+    let { script } = result
+    context.log('successfully deployed worker script\n', script)
```
Comment (EverlastingBugstopper) [inline registry/cloudflare-workers/index.js]: Why make an extra variable when you already have `result`?
```context.log(`successfully deployed worker script ${result}`)```

File: registry/cloudflare-workers/index.js [hunk 5]
```diff
@@ -26,44 +37,34 @@ const doWorkerDeploy = async ({ zone_id, script_path }) => {
...
-	const EMAIL = process.env.CLOUDFLARE_EMAIL
-	const url = `https://api.cloudflare.com/client/v4/zones/${zone_id}/workers/script`
 
-	const headers = {
-		"X-Auth-Email": `${EMAIL}`,
-		"X-Auth-Key": `${AUTH_KEY}`,
-		"Content-Type": "application/json"
-	}
+const doWorkerDeploy = async ({ zone_id, script_path }) => {
+  const AUTH_KEY = process.env.CLOUDFLARE_AUTH_KEY
```
Comment (EverlastingBugstopper) [inline registry/cloudflare-workers/index.js]: @pinkerton add some documentation so users know they must have their auth key and email set as an environment variable

`export CLOUDFLARE_EMAIL=${email}`
`export CLOUDFLARE_AUTH_KEY=${auth_key}`

File: registry/cloudflare-workers/index.js [hunk 6]
```diff
@@ -26,44 +37,34 @@ const doWorkerDeploy = async ({ zone_id, script_path }) => {
...
 }
-/*
-const remove = (input, context) => {
-	context.log('Removing worker script ')
-	await removeWorker(input)
-	context.log('awesome ')
+
+const remove = async (input, context) => {
+  context.log('removing!')
+  await removeWorker(input)
```
Comment (EverlastingBugstopper) [inline registry/cloudflare-workers/index.js]: Should add a check here to see if the fetch was successful, log errors if something goes wrong

**Review by EverlastingBugstopper — CHANGES_REQUESTED** [review]
Really great work here. Glad you got the remove figured out!

**Pull request closed without merging**